picard/ui/mainwindow.py

```diff
@@ -1210,10 +1207,11 @@ def open_folder(self):
     def _ensure_fingerprinting_configured(self, callback):
         config = get_config()
 
-        def on_finished(result):
-            callback(config.setting['fingerprinting_system'])
         if not config.setting['fingerprinting_system']:
             if self._show_analyze_settings_info():
+                def on_finished(result):
+                    callback(config.setting['fingerprinting_system'])
```

Comment:
Maybe use functools.partial:

```
on_finished = partial(callback, config.setting['fingerprinting_system'])
```

Reply:
Hmmm, actually `partial()` cannot be used like this as `on_finished()` takes one (unused) argument (`result`).

Reply:
Indeed, I missed that.